### ISSUE TITLE
Cannot Change Episode Date + More

### ISSUE THREAD

**Issue by sheriffpuma** [PodcastGenerator/PodcastGenerator#434]
I have FTP indexed an MP3 file and now I can't change the date
AH01215: PHP Warning:  touch(): Utime failed: Operation not permitted in /podcast/admin/episodes_edit.php on line 92: /usr/local/cpanel/cgi-sys/ea-php74, referer: /podcast/admin/episodes_edit.php?name=2021-09-05-example.mp3

Manually changing XML file gets erased after I try to open an episode for editing using the Pod Gen software. It seems like Pod Gen software corrupts the XML file.

The file permissions will not update in Filezilla nor cPanel, I have to delete the file after making a copy of it, rename the file after re-uploading the XML, and then I can change permissions. This has happened to my last two uploads.

The date is now being added to file name reformatting, I do not want the date added to the filename, I want to leave the filename as–is when uploading like I could do when using the older version.

I am not sure what went wrong in version 3.1.1? But I am willing to give you any server config info, suggestion, or anything else needed to help fix this software issue.

–

I also noticed these errors in a different directory where I have another copy of PodGen installed:

AH01215:   thrown in /music/components/getid3/module.audio.ac3.php on line 18: /usr/local/cpanel/cgi-sys/ea-php74

AH01215: PHP Fatal error:  Uncaught Error: Class 'getid3_handler' not found in /music/components/getid3/module.audio.ac3.php:18: /usr/local/cpanel/cgi-sys/ea-php74

**Comment by flattery103** [comment]
I can confirm the editing issue. In error logs I get:  [Sun Sep 05 15:21:44.864082 2021] [php7:warn] [pid 75324] [client 68.117.73.106:50619] PHP Warning:  touch(): Utime failed: Operation not permitted in /*****/admin/episodes_edit.php on line 92, referer: https://********/admin/episodes_edit.php?name=2021-09-05-Linux_Show_Ep0.mp3

**Comment by flattery103** [comment]
This is a permissions issue. If you uploaded the files with a user that the webserver doesn't have access to, then the webserver has no way to edit the file. The server actually updates the file's date. The fix is to set the file's permissions so the server can edit them. 

Either do a chown to the same user that the .XML files are in the media directory. Or the dirty way to do it (and less safe) is to give them global write permissions (such as chmod 766 file).

**Comment by sheriffpuma** [comment]
Do you think that maybe you can give instructions like this?
Step 1: Go here, do this
Step 2: Go there, do that

**Comment by flattery103** [comment]
I have nothing to do with the development, so I can't answer those questions. I just came across the same problem and found the cause and solution.  There are several ways to go about fixing it so I will step through one and many steps depend on your setup.

Step 1: Open a command line to your server via ssh or local consol
Step 2: Change the directory to the location of your media files. via "cd /location/to/files/media/"
Step 3: Check to see who owns the .XML files in that directory by doing "ls -al"
Step 4: Change the ower of all the files in the media directory to the same owner that owns the XML. In my example they are owned by www-data so I did "chown www-data:www-data *"


A second way would be to give them all global read/write permissions. To do this, follow steps one and two above, then do:
Step 3: "chmod 766 *"


A third way is by changing these with your ftp program such as FileZilla. Find the files with FileZilla or whatever you use and right click on the media files and set the permissions all the have global read & write permissions.

**Comment by sheriffpuma** [comment]
Thank you.

It's a shame that this software wants to suddenly start acting stupid for absolutely no reason.

The feedback after asking for the steps to fix this instead of having to try and manipulate the permissions every single time from FTP, which doesn't always work, was meant for the developers of this software.

Thanks again for your fix, even though we should not have had to do this at all in the first place, but suddenly have to do now for whatever dumb reason.

There's a new version of this coming out soon by a different development team. They will charge for their version, but I would rather a well maintained paid version than have to deal with this weak edition that looks like it was designed by a complete noob. No offense to these developers, because I have used this for years now, but I have had to do a lot of backend work in order to get this the way I needed it to be, so I'm kinda looking at this like it's a joke, but that's just my opinion.

Thank you to the development team that made this possible in the first place. But, I've definitely outgrown  this version by a long shot and I'm just waiting for this next version by the other team to come out.
